File: Data Structures & Algorithms/duplicate-integer/submission-0.rs

use std::collections::HashSet;

impl Solution {
    pub fn has_duplicate(nums: Vec<i32>) -> bool {
        let mut set: HashSet<i32> = HashSet::new();

        for x in &nums {
            if set.contains(x) {
                return true;
            } else {
                set.insert(*x);
            }
        }

        false
    }
}

File: Data Structures & Algorithms/remove-element/submission-0.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        var currentIndexValue = 0;
        for (var index = 0; index < nums.Length; index++) {
            if (nums[index] != val) {
                nums[currentIndexValue] = nums[index];
                currentIndexValue++;
            }
        }
        return currentIndexValue;
    }
}